File: AgentDash/scripts/run-migrations.mjs
#!/usr/bin/env node
/**
 * Run AgentDash migrations on Supabase
 * Uses Postgres connection from SUPABASE_DB_URL
 */
import { readFileSync, existsSync } from "fs";
import { fileURLToPath } from "url";
import { dirname, join } from "path";
import pg from "pg";

const { Client } = pg;

const __dirname = dirname(fileURLToPath(import.meta.url));
const ROOT = join(__dirname, "..");

// Load env: try AgentDash/supabase.env then parent/supabase.env (when AgentDash is in a monorepo)
function loadEnv() {
  const envPath = existsSync(join(ROOT, "supabase.env"))
    ? join(ROOT, "supabase.env")
    : join(ROOT, "..", "supabase.env");
  const content = readFileSync(envPath, "utf8");
  const env = {};
  for (const line of content.split("\n")) {
    const t = line.trim();
    if (!t || t.startsWith("#")) continue;
    const i = t.indexOf("=");
    if (i === -1) continue;
    const key = t.slice(0, i).trim();
    const val = t.slice(i + 1).trim().replace(/^["']|["']$/g, "");
    env[key] = val;
  }
  return env;
}

const env = loadEnv();
const dbUrl = env.SUPABASE_DB_URL;

if (!dbUrl) {
  console.error("SUPABASE_DB_URL not found in supabase.env");
  process.exit(1);
}

const migrations = [
  "20250203000000_initial_schema.sql",
  "20250203000001_rls_policies.sql",
  "20250203000002_seed_categories.sql",
];

async function runMigrations() {
  const client = new Client({ connectionString: dbUrl });
  await client.connect();
  console.log("Connected to database");

  try {
    for (const file of migrations) {
      const path = join(ROOT, "supabase", "migrations", file);
      const sql = readFileSync(path, "utf8");
      console.log(`\nRunning ${file}...`);
      await client.query(sql);
      console.log(`✓ ${file} completed`);
    }
    console.log("\n✅ All migrations completed successfully!");
  } catch (error) {
    console.error("\n❌ Migration failed:", error.message);
    process.exit(1);
  } finally {
    await client.end();
  }
}

runMigrations();

File: AgentDash/scripts/run_migrations.py
#!/usr/bin/env python3
"""
Run AgentDash migrations on Supabase
Uses Postgres connection from SUPABASE_DB_URL in parent supabase.env
"""
import os
import sys
from pathlib import Path

# Add parent to path to read supabase.env
ROOT = Path(__file__).resolve().parent.parent.parent
sys.path.insert(0, str(ROOT))

# Load env from parent supabase.env
def load_env():
    env_path = ROOT / "supabase.env"
    if not env_path.exists():
        print(f"supabase.env not found at {env_path}")
        sys.exit(1)
    
    env = {}
    with open(env_path) as f:
        for line in f:
            line = line.strip()
            if not line or line.startswith("#"):
                continue
            if "=" not in line:
                continue
            key, val = line.split("=", 1)
            key = key.strip()
            val = val.strip().strip('"').strip("'")
            env[key] = val
    return env

env = load_env()
db_url = env.get("SUPABASE_DB_URL")

if not db_url:
    print("SUPABASE_DB_URL not found in supabase.env")
    sys.exit(1)

try:
    import psycopg2
except ImportError:
    print("psycopg2 not installed. Install with: pip install psycopg2-binary")
    sys.exit(1)

MIGRATIONS_DIR = Path(__file__).resolve().parent.parent / "supabase" / "migrations"
migrations = [
    "20250203000000_initial_schema.sql",
    "20250203000001_rls_policies.sql",
    "20250203000002_seed_categories.sql",
]

def run_migrations():
    conn = psycopg2.connect(db_url)
    conn.autocommit = False
    cur = conn.cursor()
    
    try:
        for filename in migrations:
            filepath = MIGRATIONS_DIR / filename
            if not filepath.exists():
                print(f"Migration file not found: {filepath}")
                continue
            
            print(f"\nRunning {filename}...")
            with open(filepath) as f:
                sql = f.read()
            
            try:
                cur.execute(sql)
                conn.commit()
                print(f"✓ {filename} completed")
            except Exception as e:
                conn.rollback()
                print(f"✗ {filename} failed: {e}")
                raise
        
        print("\n✅ All migrations completed successfully!")
    finally:
        cur.close()
        conn.close()

if __name__ == "__main__":
    run_migrations()
